src/watchmaker/status/providers/aws.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+"""AWS Status Provider."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals, with_statement)
+
+import logging
+
+import watchmaker.utils as utils
+from watchmaker.conditions import HAS_BOTO3
+from watchmaker.status.providers.abstract import (AbstractStatusProvider,
+                                                  StatusProviderError)
+
+
+class AWSStatusProvider(AbstractStatusProvider):
+    """Concrete implementation of the AWS status cloud provider."""
+
+    identifier = 'aws'
+
+    def __init__(self, logger=None):
+        self.logger = logger or logging.getLogger(__name__)
+        self.metadata_id_url = (
+            'http://169.254.169.254/latest/meta-data/instance-id'
+        )
+        self.instance_id = None
+        self.initialize()
+
+    def initialize(self):
+        """Initialize instance id."""
+        if not self.instance_id:
```

Comment:
I always like to reverse these conditions, where it helps reduce indentation.

```suggestion
        if self.instance_id:
            return

        try:
```

Reply:
I can agree with that.